compiler/resolution/src/org/jetbrains/kotlin/resolve/calls/components/ResolutionParts.kt

```diff
@@ -652,6 +654,28 @@ internal object CheckReceivers : ResolutionPart() {
         resolveKotlinArgument(receiverArgument, receiverParameter, receiverInfo)
     }
 
+    private fun KotlinResolutionCandidate.searchForAdditionalReceivers(): List<SimpleKotlinCallArgument>? {
+        val result = mutableListOf<ReceiverValueWithSmartCastInfo>()
+        val candidateReceivers = scopeTower.lexicalScope.parentsWithSelf
+            .flatMap { if (it is LexicalScope) scopeTower.getImplicitReceivers(it) else emptyList() }
+        for (receiver in candidateDescriptor.additionalReceiverParameters) {
+            val expectedReceiverType = receiver.type
+            val expectedReceiverTypeClosestBound =
+                if (expectedReceiverType.isTypeParameter()) expectedReceiverType.supertypes().first()
+                    .replaceArgumentsWithStarProjections()
+                else expectedReceiverType.replaceArgumentsWithStarProjections()
+            val selectedCandidate = candidateReceivers.firstOrNull { candidateReceiver ->
+                val candidateReceiverType = candidateReceiver.receiverValue.type.replaceArgumentsWithStarProjections()
+                NewKotlinTypeChecker.Default.isSubtypeOf(candidateReceiverType, expectedReceiverTypeClosestBound)
+            } ?: run {
```

Comment:
You commented in the previous PR:

> I'd just return null here (with ?: return null), but reported an error on the call-site
So, the responsibility of the function would be limited and clear

The problem here is that I will lose the information about the missing receiver if I report an error outside of this function.